File: AC_System/app/api/containers.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from app.core.user import AdminUser
from app.services.docker_manager import docker_manager

router = APIRouter(prefix="/containers", tags=["Container Management"])

class ContainerAction(BaseModel):
    action: str

@router.get("/")
async def get_containers_list(user: AdminUser):
    """Zwraca listę wszystkich mikroserwisów HomeOS i ich status."""

    return await docker_manager.list_homeos_containers()


@router.post("/{container_name}/control")
async def control_container(container_name: str, payload: ContainerAction, user: AdminUser):
    """Uruchamia, zatrzymuje lub restartuje kontener."""

    return await docker_manager.control_container(container_name, payload.action)


@router.get("/{container_name}/logs")
async def get_logs(container_name: str, user: AdminUser, tail: int = 50):
    """Pobiera ostatnie logi z wybranego kontenera."""

    return await docker_manager.get_container_logs(container_name, tail)

@router.get("/{container_name}/stats")
async def get_container_stats(container_name: str, user: AdminUser):
    """Pobiera aktualne statystyki użycia zasobów dla danego kontenera."""
        
    return await docker_manager.get_container_stats(container_name)

File: AA_Frontend/HomeOS/src/api/system/containers.ts
import { api } from "../axiosClient";
import type { ContainerStatus } from "./types.ts";

export const getContainersStatus = async (): Promise<ContainerStatus[]> => {
  const response = await api.get(`/system/containers/`);
  return response.data;
}

export const controlContainer = async (container_name: string, action: 'start' | 'stop' | 'restart'): Promise<void> => {
  await api.post(`/system/containers/${container_name}/control/${action}`);
}

export const getContainerLogs = async (container_name: string): Promise<string[]> => {
  const response = await api.get(`/system/containers/${container_name}/logs`);
  return response.data;
}